python/cudf/cudf/core/indexed_frame.py

```diff
@@ -283,20 +288,17 @@ def _from_columns(
         If `index_names` is set, the first `len(index_names)` columns are
         used to construct the index of the frame.
         """
-        data_columns = columns
-
         n_index_columns = len(index_names) if index_names else 0
-        index_columns = columns[:n_index_columns]
-        data_columns = columns[n_index_columns:]
-
-        out = super()._from_columns(data_columns, column_names)
+        out = super()._from_columns(columns[n_index_columns:], column_names)
 
-        if index_names is not None:
-            out._index = cudf.core.index._index_from_columns(index_columns)
+        if n_index_columns:
```

Comment:
Would prefer an explicit check here.
```suggestion
        if n_index_columns > 0:
```